pyartifactory/objects.py

```diff
@@ -895,6 +896,46 @@ def download(self, artifact_path: str, local_directory_path: str = ".") -> str:
                 self._download(full_path, local_path.parent)
         return f"{local_directory_path}/{basename}"
 
+    def list(
+        self,
+        artifact_path: str,
+        recursive: bool = True,
+        depth: int = 1,
```

Comment:
What do you think about having the `depth` argument be `None` by default, and only being sent to the API if it is defined? That way if people pass the `recursive` argument they will get the full file listing by default, unless they specify a max `depth`.
An example of this implementation is in the following comment.

Also I am not sure about the default `recursive` behavior, is it better to list all files or only the first level?